pkg/antctl/raw/multicluster/get/clusterset.go

```diff
@@ -0,0 +1,200 @@
+	argsNum := len(args)
+	singleResource := false
+	if argsNum > 0 {
+		singleResource = true
+	}
+
+	k8sClient, err := raw.AddSchemeToClientCfg(kubeconfig)
+	if err != nil {
+		return err
+	}
+
+	var clusterSets []multiclusterv1alpha1.ClusterSet
+	if singleResource {
+		clusterSetName := args[0]
+		clusterSet := multiclusterv1alpha1.ClusterSet{}
+		err = k8sClient.Get(context.TODO(), types.NamespacedName{
+			Namespace: optionsClusterSet.namespace,
+			Name:      clusterSetName,
+		}, &clusterSet)
+		if err != nil {
+			return err
+		}
+		gvks, unversioned, err := k8sClient.Scheme().ObjectKinds(&clusterSet)
+		if err != nil {
+			return err
+		}
+		if !unversioned && len(gvks) == 1 {
+			clusterSet.SetGroupVersionKind(gvks[0])
+		}
+		clusterSets = append(clusterSets, clusterSet)
+	} else {
+		clusterSetList := &multiclusterv1alpha1.ClusterSetList{}
+		err = k8sClient.List(context.TODO(), clusterSetList, &client.ListOptions{Namespace: optionsClusterSet.namespace})
+		if err != nil {
+			return err
+		}
+		clusterSets = clusterSetList.Items
+	}
+
+	if len(clusterSets) == 0 {
+		if optionsClusterSet.namespace != "" {
+			fmt.Printf("No resource found in Namespace %s\n", optionsClusterSet.namespace)
+		} else {
+			fmt.Println("No resource found in all Namespaces")
+		}
+		return nil
+	}
+
+	switch optionsClusterSet.outputFormat {
+	case "json":
+		err := antctl.JsonOutput(clusterSets, os.Stdout)
+		if err != nil {
+			return err
+		}
+	case "yaml":
+		err := antctl.YamlOutput(clusterSets, os.Stdout)
+		if err != nil {
+			return err
+		}
+	default:
+		if singleResource {
+			fmt.Println(outputClusterSet(clusterSets))
+		} else {
+			fmt.Println(outputAllClusterSet(clusterSets))
+		}
+	}
+
+	return nil
+}
+
+func outputClusterSet(clusterSets []multiclusterv1alpha1.ClusterSet) string {
+	var output strings.Builder
+
+	formatter := "%-25s%-25s%-25s%-28s%-20s%-10s\n"
+	output.Write([]byte(fmt.Sprintf(formatter, "CLUSTERID", "NAMESPACE", "CLUSTERSETID", "TYPE", "STATUS", "REASON")))
+	sort.SliceStable(clusterSets, func(i, j int) bool {
+		return strings.Compare(clusterSets[i].Namespace, clusterSets[j].Namespace) == 1
+	})
+	for _, clusterSet := range clusterSets {
+		sort.SliceStable(clusterSet.Status.ClusterStatuses, func(i, j int) bool {
+			return strings.Compare(clusterSet.Status.ClusterStatuses[i].ClusterID, clusterSet.Status.ClusterStatuses[j].ClusterID) == 1
+		})
+		for _, r := range clusterSet.Status.ClusterStatuses {
+			for _, condition := range r.Conditions {
+				fmt.Fprintf(&output, formatter, r.ClusterID, clusterSet.Namespace, clusterSet.Name, condition.Type, condition.Status, condition.Reason)
+			}
+		}
+	}
+
+	return output.String()
+}
+
+func outputAllClusterSet(clusterSets []multiclusterv1alpha1.ClusterSet) string {
+	var output strings.Builder
+	formatter := "%-25s%-20s%-20s%-20s%-25s%-10s\n"
+	output.Write([]byte(fmt.Sprintf(formatter, "CLUSTERSETID", "NAMESPACE", "TOTAL_CLUSTERS", "READY_CLUSTERS", "OVERALL_CONDITION", "REASON")))
```

Comment:
I suggested earlier - it is strange to output with two different formats for the same client. What you think we just output each ClusterSet the same way as outputting a single ClusterSet (e.g. calling outputClusterSet() for each ClusterSet with your current implementation), and add a new line between two ClusterSets?

Reply:
@jianjuns Oh Thanks, Jianjun, this part was discussed [https://github.com/antrea-io/antrea/pull/3287#discussion_r807358541](url) , the thought we had was that for the overall output we did not need to output the details of each cluster, just some overall information(like Total clusters, ready clusters). This means that outputClusterSet has everything outputAllClusterSet has, and provides more information. So we don't need to add some new format here, I will just provide the detail information here to avoid confusing customer with different output of clusterset command.

Reply:
But what you think about my idea: "output each ClusterSet the same way as outputting a single ClusterSet (e.g. calling outputClusterSet() for each ClusterSet with your current implementation), and add a new line between two ClusterSets"? Do you see any issue with that?

I feel thtat is consistent with yaml and json outputs.